src/gdbstub_impl/ext/reverse_exec.rs

```diff
@@ -0,0 +1,106 @@
+                let mut check_gdb_interrupt = || match res.as_conn().peek() {
+                    Ok(Some(0x03)) => true, // 0x03 is the interrupt byte
+                    Ok(Some(_)) => false,   // it's nothing that can't wait...
+                    Ok(None) => false,
+                    Err(e) => {
+                        err = Err(Error::ConnectionRead(e));
+                        true // break ASAP if a connection error occurred
+                    }
+                };
+
+                let stop_reason = match target.base_ops() {
+                    BaseOps::MultiThread(ops) => {
+                        if let Some(ops) = ops.support_reverse_cont() {
+                            ops.reverse_cont(&mut check_gdb_interrupt)
+                                .map_err(Error::TargetError)?
+                        } else {
+                            return Err(Error::PacketUnexpected);
+                        }
+                    }
+
+                    BaseOps::SingleThread(_ops) => {
+                        return Err(Error::PacketUnexpected);
+                    }
+                };
+
+                err?;
+
+                // FIXME: Unnecessary unwrap
+                self.finish_exec(res, target, stop_reason)?.unwrap()
+            }
+        };
+
+        Ok(handler_status)
+    }
+
+    // FIXME: De-duplicate with above code?
+    pub(crate) fn handle_reverse_step(
+        &mut self,
+        res: &mut ResponseWriter<C>,
+        target: &mut T,
+        command: ReverseStep,
+    ) -> Result<HandlerStatus, Error<T::Error, C::Error>> {
+        let handler_status = match command {
+            ReverseStep::bs(_) => {
+                let tid = match self.current_resume_tid {
+                    // N.B: Can't single-step all cores.
+                    SpecificIdKind::All => return Err(Error::PacketUnexpected),
+                    SpecificIdKind::WithId(tid) => tid,
+                };
+
+                // FIXME: This block is duplicated from the vCont code.
+                let mut err = Ok(());
+                let mut check_gdb_interrupt = || match res.as_conn().peek() {
+                    Ok(Some(0x03)) => true, // 0x03 is the interrupt byte
+                    Ok(Some(_)) => false,   // it's nothing that can't wait...
+                    Ok(None) => false,
+                    Err(e) => {
+                        err = Err(Error::ConnectionRead(e));
+                        true // break ASAP if a connection error occurred
+                    }
+                };
+
+                let stop_reason = match target.base_ops() {
+                    BaseOps::MultiThread(ops) => {
+                        if let Some(ops) = ops.support_reverse_step() {
+                            ops.reverse_step(tid, &mut check_gdb_interrupt)
+                                .map_err(Error::TargetError)?
+                        } else {
+                            return Err(Error::PacketUnexpected);
```

Comment:
`return Err(Error::PacketUnexpected)` is _not_ correct here, and should be `return Ok(HandlerStatus::Handled)`. The GDB protocol allows clients to send packets the target doesn't understand / implement, and the correct response is to simply ignore the packet and send back nothing.